recipes/blasfeo/all/conanfile.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from conan import ConanFile
+from conan.tools.files import (
+    apply_conandata_patches,
+    copy,
+    export_conandata_patches,
+    get,
+)
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+
+required_conan_version = ">=1.57.0"
+
+
+class BlasfeoConan(ConanFile):
+    name = "blasfeo"
+    description = "Basic linear algebra subroutines for embedded optimization."
+    topics = (
+        "blasfeo",
+        "blas",
+        "linear-algebra",
+        "matrices",
+        "embedded",
+        "optimization",
+    )
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/giaf/blasfeo"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "library_type": ["static_library", "shared_library"],
```

Comment:
```suggestion
        "shared": [True, False],
```
please, use the recommended form: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conanfile_attributes.md#options

Reply:
Hi @uilianries, thanks for your comments.

I use `make` instead of `CMake` because [per documentation](https://github.com/giaf/blasfeo#supported-computer-architectures) not all architectures are supported in CMake: *Note that the X64_INTEL_SKYLAKE_X, ARMV8A_APPLE_M1, ARMV8A_ARM_CORTEX_A76, ARMV8A_ARM_CORTEX_A73, ARMV8A_ARM_CORTEX_A55, X86_AMD_JAGUAR and X86_AMD_BARCELONA architectures are not currently supported by the CMake build system and can only be used through the included Makefile.*

When [building with make](https://blasfeo.syscop.de/docs/install/#compilation), arguments are either `static_library` or  `shared_library`, so I thought of setting the options directly without using e.g. *if* conditions in the recipe.
Plus, other arguments `LA` and `MF` can be set in the same way.

Reply:
I still don't get about the architecture, you are actually cross-building?

> static_library or shared_library, so I thought of setting the options directly without using e.g. if conditions in the recipe.

Please, keep `shared` because it's automatically recognized by Conan and hooks.

Reply:
> I still don't get about the architecture, you are actually cross-building?

Yes, I need to build the library for different architectures and configurations.
Isn't it the case that library specific flags (in this case `target`, `lf` and `mf`) should go to options? Or I don't understand the difference between settings and options correctly.

> Please, keep `shared` because it's automatically recognized by Conan and hooks.

Ok, I can change to `shared` and specify make argument based on that.